PublicMapping/districtbuilder
# Add zoom geolevel restrictions and auto-saving when switching geolevels

## Overview
Add zoom geolevel restrictions and auto-saving when switching geolevels

### Demo
![db_selector_constraints](https://user-images.githubusercontent.com/2926237/88334111-3ec44b80-ccff-11ea-9ff0-320a8363e98d.gif)

### Notes
There is a small amount of weirdness introduced in `reducers/districtDrawing.ts` where we need state held in `reducers/projectData.ts` to call the API function to save the updated district definition, so instead we kick off an action, passing the missing data along to the reducer that has the rest of the data to make the API call (previously we were getting around this by having the necessary data where the action is dispatched at the call site but now it needs to be called in two places to accommodate the auto-saving). I suppose this is the price we pay for encapsulation, but it does add complexity. The other alternative is to recombine the two reducers into one.

FWIW I still think the auto-saving is confusing and would rather just not let the user switch geolevels with some messaging if they have a selection instead of auto-saving for them.

## Testing Instructions
* Make sure you can only select the block level after zooming in enough to see blocks
* Make sure once you select a block, you can't zoom out so far as to no longer be able to see blocks
* Switch geolevels and make sure your select was saved

Closes #201
